test/test_jit.py

```diff
@@ -1199,37 +1199,61 @@ def test_quant_fusion(self):
         return (%r_dequant)""",
             # addmm -> quantized::linear
             """
-graph(%a, %w, %b, %a_scale, %a_zero_point, %a_dtype, %w_scale, %w_zero_point, %w_dtype, %r_scale, %r_zero_point, %r_dtype, %4):
+graph(%self, %a, %a_scale, %a_zero_point, %a_dtype, %r_scale, %r_zero_point, %r_dtype, %4):
         %a_quant = aten::quantize_per_tensor(%a, %a_scale, %a_zero_point, %a_dtype)
         # CHECK-NOT: aten::dequantize
         %a_dequant = aten::dequantize(%a_quant)
-        %w_quant = aten::quantize_per_tensor(%w, %w_scale, %w_zero_point, %w_dtype)
+        %packed_params = prim::GetAttr[name="_packed_params"](%self)
```

Comment:
This test is over-specialized and thus too fragile and not very helpful. It just replicates the pattern almost exactly - what's the point of having it? It doesn't check if the pattern is correct. At most, it will check that subgraph rewriter works, but it has its own tests. The test should not check implementation details (like `# CHECK-NOT: prim::TupleIndex`), it should rather check that the pass is doing what it's supposed to do - in this case replace `addmm`  with `quantized::linear`. For that two CHECK/CHECK_NOT directives would be enough.

Reply:
OK

Reply:
But to test it with actual patterns we see we might need e2e test

Reply:
Yes, we definitely need an e2e test.

Reply:
many `CHECK-NOT` is part of the pattern as well, `CHECK-NOT`s are making sure we actually removed those nodes

Reply:
If you find `quantized::linear` in the output then you know that the pattern was matched. There is no reason the check other details of the pattern - or you will be just copying the pattern into the test. If the pattern is buggy, you would just copy the buggy part into the test. This test cannot check if the pattern is buggy (that's the job of e2e test), and there is no need to check if the pattern rewrite was accurate (that's covered by tests for subgraph rewriter).